### PR TITLE
Update to TypeScript 5.0

### PR DESCRIPTION
**Pull request by rictic**

### CODE REVIEW

**Comment by rictic** [comment]
.d.ts changes look chill: https://gist.github.com/rictic/1ddd4afd56b88f008609d05f05e4ce80

File: package.json [hunk 1]
```diff
@@ -206,6 +206,7 @@
     "@web/test-runner-mocha": "^0.7.5",
     "@web/test-runner-playwright": "^0.9.0",
     "@web/test-runner-saucelabs": "^0.8.0",
+    "chai": "^4.3.7",
```
Comment (justinfagnani) [inline package.json:209]: why are we adding `chai`? Does it pull along types? There's also `@types/chai`

File: packages/labs/context/package.json [hunk 2]
```diff
@@ -147,6 +148,7 @@
     "lit": "^2.7.0"
   },
   "devDependencies": {
+    "@esm-bundle/chai": "4.3.4",
```
Comment (justinfagnani) [inline packages/labs/context/package.json:151]: this with `"@esm-bundle/chai": "4.3.4-fix.0"` will give us multiple copies. Can we standardize on one version?

File: packages/localize-tools/src/modes/transform.ts [hunk 3]
```diff
@@ -218,17 +221,17 @@ class Transformer {
           '_LIT_LOCALIZE_CONFIGURE_TRANSFORM_LOCALIZATION_'
         )
       ) {
-        return ts.createObjectLiteral(
+        return factory.createObjectLiteralExpression(
```
Comment (justinfagnani) [inline packages/localize-tools/src/modes/transform.ts:224]: Did this API exist on typescript@4.7? Checking if this is a breaking change for localize-tools.

File: packages/ts-transformers/package.json [hunk 4]
```diff
@@ -57,7 +57,7 @@
   },
   "dependencies": {
     "ts-clone-node": "^1.0.0",
-    "typescript": "~4.7.4"
+    "typescript": "~5.0.0"
```
Comment (justinfagnani) [inline packages/ts-transformers/package.json:60]: Is this a breaking change? Thinking about lit.dev here.

File: packages/localize-tools/src/modes/transform.ts (hunk 3 above)
Comment (jpzwarte) [reply packages/localize-tools/src/modes/transform.ts:224]: If it is, we probably need https://github.com/lit/lit/pull/3800, but with `~5.0.0` cc @aomarks

Comment (jpzwarte) [reply packages/localize-tools/src/modes/transform.ts:224]: Also, if it *is* compatible, then we need to add `~5.0.0` (see https://github.com/lit/lit/pull/3800)

Comment (rictic) [reply packages/localize-tools/src/modes/transform.ts:224]: There's breaking changes in the TS APIs, so will do this as a breaking change to localize-tools and ts-transformers

File: packages/ts-transformers/package.json (hunk 4 above)
Comment (rictic) [reply packages/ts-transformers/package.json:60]: Yeah, marked as such

File: package.json (hunk 1 above)
Comment (rictic) [reply package.json]: Was trying to fix a lint error but ended up going with another approach, removed

File: packages/labs/context/package.json (hunk 2 above)
Comment (rictic) [reply packages/labs/context/package.json]: Done

File: package.json (hunk 1 above)
Comment (rictic) [reply package.json]: Was trying to fix a lint error but ended up going with another approach, removed

File: packages/labs/context/package.json (hunk 2 above)
Comment (rictic) [reply packages/labs/context/package.json]: Done

**Review by justinfagnani — APPROVED** [review]

**Pull request merged**